code/modules/mob/living/carbon/human/species_types/beefmen.dm

```diff
@@ -105,25 +105,37 @@
 
 
 
-/mob/living/carbon/proc/ReassignForeignBodyparts()
-	if (get_bodypart(BODY_ZONE_HEAD)?.type != part_default_head)  // <----- I think :? is used for procs instead of .? ...but apparently BYOND does that swap for you. //(!istype(get_bodypart(BODY_ZONE_HEAD), part_default_head))
-		qdel(get_bodypart(BODY_ZONE_HEAD))
-		new part_default_head().replace_limb(src,TRUE)
-	if (get_bodypart(BODY_ZONE_CHEST)?.type != part_default_chest)
-		qdel(get_bodypart(BODY_ZONE_CHEST))
-		new part_default_chest().replace_limb(src,TRUE)
-	if (get_bodypart(BODY_ZONE_L_ARM)?.type != part_default_l_arm)
-		qdel(get_bodypart(BODY_ZONE_L_ARM))
-		new part_default_l_arm().replace_limb(src,TRUE)
-	if (get_bodypart(BODY_ZONE_R_ARM)?.type != part_default_r_arm)
-		qdel(get_bodypart(BODY_ZONE_R_ARM))
-		new part_default_r_arm().replace_limb(src,TRUE)
-	if (get_bodypart(BODY_ZONE_L_LEG)?.type != part_default_l_leg)
-		qdel(get_bodypart(BODY_ZONE_L_LEG))
-		new part_default_l_leg().replace_limb(src,TRUE)
-	if (get_bodypart(BODY_ZONE_R_LEG)?.type != part_default_r_leg)
-		qdel(get_bodypart(BODY_ZONE_R_LEG))
-		new part_default_r_leg().replace_limb(src,TRUE)
+/mob/living/carbon/proc/ReassignForeignBodyparts() //This proc hurts me so much, it used to be worse, this really should be a list or something
+	var/obj/item/bodypart/head = get_bodypart(BODY_ZONE_HEAD)
+	if (head?.type != part_default_head)  // <----- I think :? is used for procs instead of .? ...but apparently BYOND does that swap for you. //(!istype(get_bodypart(BODY_ZONE_HEAD), part_default_head))
+		qdel(head)
+		var/obj/item/bodypart/limb = new part_default_head
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/chest = get_bodypart(BODY_ZONE_CHEST)
+	if (chest?.type != part_default_chest)
+		qdel(chest)
+		var/obj/item/bodypart/limb = new part_default_chest
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/l_arm = get_bodypart(BODY_ZONE_L_ARM)
+	if (l_arm?.type != part_default_l_arm)
+		qdel(l_arm)
+		var/obj/item/bodypart/limb = new part_default_l_arm
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/r_arm = get_bodypart(BODY_ZONE_R_ARM)
+	if (r_arm?.type != part_default_r_arm)
+		qdel(r_arm)
+		var/obj/item/bodypart/limb = new part_default_r_arm
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/l_leg = get_bodypart(BODY_ZONE_L_LEG)
+	if (l_leg?.type != part_default_l_leg)
+		qdel(l_leg)
+		var/obj/item/bodypart/limb = new part_default_l_leg
+		limb.replace_limb(src,TRUE)
+	var/obj/item/bodypart/r_leg = get_bodypart(BODY_ZONE_R_LEG)
+	if (r_leg?.type != part_default_r_leg)
+		qdel(r_leg)
+		var/obj/item/bodypart/limb = new part_default_r_leg
+		limb.replace_limb(src,TRUE)
```

Comment:
This proc should be an assoc list, but I just... I don't want to. It's longer, but it doesn't throw warnings anymore